lib/ddtrace/version.rb

```diff
@@ -10,5 +10,9 @@ module VERSION
     # Support for Ruby < 2.1 is currently deprecated in the tracer.
     # Support will be dropped in the near future.
     MINIMUM_RUBY_VERSION = '2.0.0'.freeze
+
+    # Ruby 3.2 is not supported, given kwargs transition and disappearance of ruby2_keywords
```

Comment:
I haven't been able to find information about the removal of `ruby2_keywords` in Ruby 3.2, could you point me to any sources on it?

Reply:
See this comment: https://github.com/DataDog/dd-trace-rb/pull/1495#issuecomment-833435661

I also [asked Benoit Daloze](https://twitter.com/eregontp/status/1390390407457034243), TruffleRuby lead and one of the regularly involved persons in the kwargs subject on [bugs.ruby-lang.org](https://bugs.ruby-lang.org/users/772):

>> Given that `respond_to?(:ruby2_keywords)` would become false around Ruby 2.6 EOL, wouldn't the correct AND future-proof way be to `if RUBY_VERSION < '3'` that `*args` section, and use `*args, **kwargs` on the `else` clause?
>
> Yes, that's technically correct, so this would work:
>
> ```
> if RUBY_VERSION < "3.0"
>   def foo(*args, &block)
>     target(*args, &block)
>   end
>   ruby2_keywords :foo if respond_to?(:ruby2_keywords, true)
> else
>   def foo(*args, **kwargs, &block)
>     target(*args, **kwargs, &block)
>   end
> end
> ```
>
> However, the idea of ruby-core is that `ruby2_keywords` would only be removed when 2.6 is EOL and unsupported by most gems, or in other words when gems can assume Ruby >= 3.0, and then only the second part is needed. That strategy is really about avoiding the duplicated code.
>
> So ruby-core's view is change (*args) to (*args)+ruby2_kewords so it works on Ruby 2.7+, and later on (when your app/gem can assume Ruby 3+) change (*args)+ruby2_kewords to (*args, **kwargs).

Not sure if Benoit is core but he's at least core contributor and very much in touch. Anyway, I figured out, I could just as well [ask Matz](https://twitter.com/lloeki/status/1390651699958882304) too for a clarification on the timetable.